aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -687,6 +692,250 @@ public void onError(Exception error) {
+                                                        AuthUpdateAttributeStep.
+                                                                DONE,
+                                                        Collections.emptyMap(),
+                                                        null)
+                                        ));
+                                onSuccess.accept(resultMap);
+                            }
+
+                            for (UserCodeDeliveryDetails details : result) {
+                                if (details.getAttributeName().equals(attributeKey)) {
+                                    Map<AuthUserAttributeKey, AuthUpdateAttributeResult> resultMap =
+                                            new HashMap<>();
+                                    resultMap.put(new AuthUserAttributeKey(attributeKey),
+                                            new AuthUpdateAttributeResult(
+                                                    true,
+                                                    new AuthNextUpdateAttributeStep(
+                                                            AuthUpdateAttributeStep.
+                                                                    CONFIRM_ATTRIBUTE_WITH_CODE,
+                                                            Collections.emptyMap(),
+                                                            convertCodeDeliveryDetails(details))
+                                            ));
+                                    onSuccess.accept(resultMap);
+                                } else {
+                                    onError.accept(new AuthException(
+                                            "Returned an incorrect attribute name",
+                                            "See attached exception for more details"));
+                                }
+                            }
+                        }
+                    }
+
+                    @Override
+                    public void onError(Exception error) {
+                        onError.accept(new AuthException(
+                                "Failed to update user attributes",
+                                error,
+                                "See attached exception for more details"
+                        ));
+                    }
+                });
+    }
+
+    @Override
+    public void resendUserAttributeConfirmationCode(
+            @NonNull AuthUserAttributeKey attributeKey,
+            @NonNull Consumer<AuthCodeDeliveryDetails> onSuccess,
+            @NonNull Consumer<AuthException> onError
+    ) {
+        String attributeName = attributeKey.getKeyString();
+        awsMobileClient.verifyUserAttribute(attributeName, new Callback<UserCodeDeliveryDetails>() {
+            @Override
+            public void onResult(UserCodeDeliveryDetails result) {
+                if (result.getAttributeName().equals(attributeName)) {
```

Comment:
So I actually think a better option here is to remove this check. If it comes back with a successful response, it means it recognized the attribute you sent and resent the code for it. You can simply return the converted response without checking this.